pkg/controller/servicebindingrequest/planner.go

```diff
@@ -78,6 +78,12 @@ var EmptyBackingServiceSelectorsErr = errors.New("backing service selectors are
 // Plan by retrieving the necessary resources related to binding a service backend.
 func (p *Planner) Plan() (*Plan, error) {
 	ns := p.sbr.GetNamespace()
+
+	var emptyApplication v1alpha1.ApplicationSelector
+	if p.sbr.Spec.ApplicationSelector == emptyApplication {
+		return nil, EmptyApplicationSelectorErr
```

Comment:
@DhritiShikhar  If ApplicationSelector is empty, would the control go ahead and generate the binding secret ?

Reply:
Yes Shoubhik.. I will fix this

Reply:
Thanks, from a user's perspective, this change nullifies the work that was done for generic secret.

Reply:
PR -> https://github.com/redhat-developer/service-binding-operator/pull/432